src/Microsoft.Health.Fhir.Core/Features/Operations/Export/CancelExportRequestHandler.cs

```diff
@@ -56,26 +61,38 @@ public async Task<CancelExportResponse> Handle(CancelExportRequest request, Canc
                 throw new UnauthorizedFhirActionException();
             }
 
-            return await _retryPolicy.ExecuteAsync(async () =>
+            CancelExportResponse cancelResponse;
+            try
             {
-                ExportJobOutcome outcome = await _fhirOperationDataStore.GetExportJobByIdAsync(request.JobId, cancellationToken);
-
-                // If the job is already completed for any reason, return conflict status.
-                if (outcome.JobRecord.Status.IsFinished())
+                cancelResponse = await _retryPolicy.ExecuteAsync(async () =>
                 {
-                    return new CancelExportResponse(HttpStatusCode.Conflict);
-                }
+                    ExportJobOutcome outcome = await _fhirOperationDataStore.GetExportJobByIdAsync(request.JobId, cancellationToken);
+
+                    // If the job is already completed for any reason, return conflict status.
+                    if (outcome.JobRecord.Status.IsFinished())
+                    {
+                        return new CancelExportResponse(HttpStatusCode.Conflict);
+                    }
+
+                    // Try to cancel the job.
+                    outcome.JobRecord.Status = OperationStatus.Canceled;
+                    outcome.JobRecord.CanceledTime = Clock.UtcNow;
 
-                // Try to cancel the job.
-                outcome.JobRecord.Status = OperationStatus.Canceled;
-                outcome.JobRecord.CanceledTime = Clock.UtcNow;
+                    outcome.JobRecord.FailureDetails = new JobFailureDetails(Resources.UserRequestedCancellation, HttpStatusCode.NoContent);
 
-                outcome.JobRecord.FailureDetails = new JobFailureDetails(Resources.UserRequestedCancellation, HttpStatusCode.NoContent);
+                    _logger.LogInformation($"Attempting to cancel export job {request.JobId}");
+                    await _fhirOperationDataStore.UpdateExportJobAsync(outcome.JobRecord, outcome.ETag, cancellationToken);
 
-                await _fhirOperationDataStore.UpdateExportJobAsync(outcome.JobRecord, outcome.ETag, cancellationToken);
+                    return new CancelExportResponse(HttpStatusCode.Accepted);
+                });
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning($"Unable to cancel export job {request.JobId}. Exception: {ex}");
```

Comment:
>LogWarning [](start = 24, length = 10)

That would only give you last exception in retry cycle( at least if I correctly understand how retry policy works)
There is chance you run retry few times and it fail with actual problem, but last time you would get just 429 because you send to much requests to DB.

Also LogError can take your exception and maybe format it in better way. Plus exception feels like error thingy than just a warning.

Reply:
In this scenario, retry only kicks in if a `JobConflictException` is thrown. So it should be enough if we log the exception that gets us out of the retry loop. Also, we'll know if we are retrying because I added a log for each attempt. 

I'll update the code to use `LogError` instead of `LogWarning`.

Reply:
Well, you call `GetExportJobByIdAsync` and `UpdateExportJobAsync` and from my perspective both of them can throw something, but I'm not expert in export jobs workflow :)